### ISSUE TITLE
Fails with IPython

### ISSUE THREAD

**Issue by eulerfan271** [wookayin/python-imgcat#11]
In an IPython (or Jupyter) console, imgcat errors and fails to load images, even when using kitty and explicitly setting the TERM environment variable to `xterm-kitty`.  I'm not sure if this a bug report or feature request (or even a problem with Jupyter!), especially since the IPython/Jupyter console explicitly changes the TERM environment variable.   I think this would be a useful feature/fix in any case, as an IPython provides many benefits over a python REPL.  

Steps to reproduce:
-Start an IPython or Jupyter console using kitty (problem occurs with both python 3.6 and 3.8 for me)
-Explicitly change TERM value with `%set_env TERM=xterm-kitty` [or don't, doesn't make a difference in the outcome :)]
-Set matplotlib backend to imgcat, and plot something
-imgcat throws `'OutStream' object has no attribute 'buffer'`

The module is working perfectly for me in kitty in a python REPL.  Thanks for this great project!

**Comment by matteha** [comment]
If this could be fixed, that would open up great possibilities for data science.

**Comment by wookayin** [comment]
Yes, I agree we need to detect ipython as a runtime and make the `imgcat()` function work properly. For now, you can use a ipython magic:

```
%load_ext imgcat
%imgcat (a python image object)
```
